zerver/lib/bugdown/__init__.py

```diff
@@ -749,8 +749,9 @@ def youtube_id(self, url: str) -> Optional[str]:
         youtube_re = r'^({schema_re}?{host_re}{param_re}?)?{id_re}(?(1).+)?$'
         youtube_re = youtube_re.format(schema_re=schema_re, host_re=host_re, id_re=id_re, param_re=param_re)
         match = re.match(youtube_re, url)
-        # URLs of the form youtube.com/playlist?list=<list-id> are incorrectly matched
-        if match is None or match.group(2) == 'playlist':
+        # URLs of the forms youtube.com/playlist?list=<list-id> and
+        # youtube.com/channel/<channel-id> are incorrectly matched
+        if match is None or match.group(2) == 'playlist' or match.group(2) == 'channel':
```

Comment:
Please add a test for this :)

Reply:
Why do we block specific words here? It would be more accurate to distinguish based on domain.

Working URLs:

* https://youtu.be/dQw4w9WgXcQ
* https://www.youtube.com/watch?v=dQw4w9WgXcQ
* https://www.youtube.com/embed/dQw4w9WgXcQ
* https://www.youtube.com/playlist?list=PLlaN88a7y2_plecYoJxvRFTLHVbIVAOoc
* https://www.youtube.com/channel/UCuAXFkgsw1L7xaCfnd5JJOw

Not working URLs:

* https://www.youtube.com/dQw4w9WgXcQ
* https://youtu.be/watch?v=dQw4w9WgXcQ
* https://youtu.be/embed/dQw4w9WgXcQ
* https://youtu.be/playlist?list=PLlaN88a7y2_plecYoJxvRFTLHVbIVAOoc
* https://youtu.be/channel/UCuAXFkgsw1L7xaCfnd5JJOw

 `youtu.be/(.*)` is nothing more than a dumb redirect to `youtube.com/watch?v=\1`, and we should not be trying to treat the two domains as equivalent. When not treating the domains as equivalent, a video ID can never be put in the same position as `playlist` or `channel`.

Reply:
I think there's a history that https://www.youtube.com/dQw4w9WgXcQ was a format that one needed to support for videos in like 2014, but there's no reason to support it now.

BTW I think we should write a unit test for this `youtube_id` function with all of those cases in it, since that's a really simple test to write, and we'd save some work.

(Though, @andersk, I also wonder if we can just delete this code and use the Open Graph stuff for Youtube with similarly good results?)